node/service/src/parachains_db/upgrade.rs

```diff
@@ -49,26 +50,39 @@ impl From<Error> for io::Error {
 }
 
 /// Try upgrading parachain's database to the current version.
-pub fn try_upgrade_db(db_path: &Path) -> Result<(), Error> {
+pub(crate) fn try_upgrade_db(db_path: &Path, db_kind: DatabaseKind) -> Result<(), Error> {
 	let is_empty = db_path.read_dir().map_or(true, |mut d| d.next().is_none());
 	if !is_empty {
-		match current_version(db_path)? {
-			0 => migrate_from_version_0_to_1(db_path)?,
-			CURRENT_VERSION => (),
-			v => return Err(Error::FutureVersion { current: CURRENT_VERSION, got: v }),
+		match get_db_version(db_path)? {
+			// 0 -> 1 migration
+			Some(0) => migrate_from_version_0_to_1(db_path, db_kind)?,
+			// Already at current version, do nothing.
+			Some(CURRENT_VERSION) => (),
+			// This is an arbitrary future version, we don't handle it.
+			Some(v) => return Err(Error::FutureVersion { current: CURRENT_VERSION, got: v }),
+			// No version file. For `RocksDB` we dont need to do anything.
+			None if db_kind == DatabaseKind::RocksDB => (),
+			// No version file. `ParityDB` did not previously have a version defined.
+			// We handle this as a `0 -> 1` migration.
+			None if db_kind == DatabaseKind::ParityDB =>
+				migrate_from_version_0_to_1(db_path, db_kind)?,
+			// Fresh db.
```

Comment:
what do you mean by "Fresh db"? :) this is unreachable

Reply:
lol